jdat_notebooks/ifu_optimal/ifu_optimal.ipynb

```diff
@@ -0,0 +1,1141 @@
+    "#Load with astropy.fits.open\n",
+    "with fits.open(filename, memmap=False) as hdulist:\n",
+    "    sci = hdulist['SCI'].data\n",
+    "    err = hdulist['ERR'].data\n",
+    "    w = wcs.WCS(hdulist[1].header)\n",
+    "    hdr = hdulist[1].header\n",
+    "    hdulist.info()\n",
+    "    print(w)\n",
+    "    \n",
+    "#Load with Spectrum1D    \n",
+    "spec1d = Spectrum1D.read(filename)\n",
+    "\n",
+    "#Wavelengths\n",
+    "wavelength = np.array(spec1d.spectral_axis.value)\n",
+    "print(wavelength)\n",
+    "\n",
+    "#Sum over spaxels\n",
+    "fnu_sum = np.sum(spec1d.flux, axis=(0, 1))\n",
+    "\n",
+    "#List of cube slices for aperture photometry\n",
+    "data = []\n",
+    "var = []\n",
+    "spec1d_len = len(spec1d.spectral_axis.value)\n",
+    "for idx in range(spec1d_len): \n",
+    "    data.append(sci[idx, :, :])\n",
+    "    var.append(err[idx, :, :])  # variance = err, not variance = err**2.  Squaring the err gives noisy results. \n",
+    "\n",
+    "#Window data and variance (and replace NaNs)\n",
+    "#The existing JWST pipeline window is overgenerous (39x33 instead of the nominal 30x30 pixels)\n",
+    "data_win = np.nan_to_num(np.array(data)[:, 5:-4, 3:])\n",
+    "data_var = np.nan_to_num(np.array(var)[:, 5:-4, 3:]) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Developer Note:** Can we fix or suppress this AsdfWarning?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Developer Note:**  Is there a way to read in with only Spectrum1D to perform all of our cube operations, not using fits.open()?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<WCS(output_frame=SpectralFrame, input_frame=CoordinateFrame, forward_transform=Model: SpectralTabular1D\n",
+       "N_inputs: 1\n",
+       "N_outputs: 1\n",
+       "Parameters: \n",
+       "  points: (array([   0,    1,    2, ..., 2056, 2057, 2058]),)\n",
+       "  lookup_table: [0.96652146 0.96675646 0.96699146 ... 1.44968146 1.44991646 1.45015146] um\n",
+       "  method: linear\n",
+       "  fill_value: nan\n",
+       "  bounds_error: True)>"
+      ]
+     },
+     "execution_count": 24,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from gwcs import WCS as GWCS\n",
+    "#How to retrieve wcs summary from spec1d\n",
+    "isinstance(spec1d.wcs, GWCS)\n",
+    "\n",
+    "#dir(spec1d.wcs)\n",
+    "spec1d.wcs.low_level_wcs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Spectral Collection option\n",
+    "flux = u.Quantity(np.random.sample((5, 10)), unit='Jy')\n",
+    "spectral_axis = u.Quantity(np.arange(50).reshape((5, 10)), unit='AA')\n",
+    "wcs = np.array([gwcs_from_array(x) for x in spectral_axis])\n",
```

Comment:
`gwcs_from_array` was never defined nor imported

Reply:
Deleted this option.  Latest specutils has new functionality to manipulate spectral cubes.